tests/k8st/tests/test_simple.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (c) 2018 Tigera, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import subprocess
+from time import sleep
+
+from kubernetes import client
+
+from tests.k8st.test_base import TestBase
+
+_log = logging.getLogger(__name__)
+_log.setLevel(logging.DEBUG)
+
+
+class TestAllRunning(TestBase):
+    def check_pod_status(self, ns):
+        pods = self.cluster.list_namespaced_pod(ns)
+
+        for pod in pods.items:
+            self.cluster.read_namespaced_pod_status(namespace=ns, name=pod.metadata.name)
+            assert pod.status.phase == 'Running'
+            _log.debug("%s\t%s\t%s", pod.metadata.name, pod.metadata.namespace, pod.status.phase)
+
+    def test_kubesystem_pods_running(self):
+        self.check_pod_status('kube-system')
+
+    def test_default_pods_running(self):
+        self.check_pod_status('default')
+
+    def test_calico_monitoring_pods_running(self):
+        self.check_pod_status('calico-monitoring')
+
+
+class TestSimplePolicy(TestBase):
+    @classmethod
+    def setUpClass(cls):
+        cluster = cls.k8s_client()
+        cluster.create_namespace(client.V1Namespace(metadata=client.V1ObjectMeta(name="policy-demo")))
+        container = client.V1Container(
+            name="nginx",
+            image="nginx:1.7.9",
+            ports=[client.V1ContainerPort(container_port=80)])
+        # Create and configure a spec section
+        template = client.V1PodTemplateSpec(
+            metadata=client.V1ObjectMeta(labels={"app": "nginx"}),
+            spec=client.V1PodSpec(containers=[container]))
+        # Create the specification of deployment
+        spec = client.ExtensionsV1beta1DeploymentSpec(
+            replicas=2,
+            template=template)
+        # Instantiate the deployment object
+        nginx_deployment = client.ExtensionsV1beta1Deployment(
+            api_version="extensions/v1beta1",
+            kind="Deployment",
+            metadata=client.V1ObjectMeta(name="nginx"),
+            spec=spec)
+        api_response = client.ExtensionsV1beta1Api().create_namespaced_deployment(
+            body=nginx_deployment,
+            namespace="policy-demo")
```

Comment:
It's hard to read what's going on with the calls above.

- If it's just "kubectl run nginx --namespace policy-demo", can we just do that instead?

- Alternatively, move into a subroutine called run_nginx or something.

Reply:
Yeah - in general, the kubernetes API calls are all much nastier than simply running the kubectl equivalent :)